app/src/main/java/com/hedvig/app/feature/genericauth/GenericAuthActivity.kt

```diff
@@ -26,24 +23,20 @@ class GenericAuthActivity : BaseActivity() {
 
     setContent {
       val viewState by model.viewState.collectAsState()
-      LaunchedEffect(model.eventsFlow) {
-        repeatOnLifecycle(Lifecycle.State.STARTED) {
-          model.eventsFlow.collect { event ->
-            when (event) {
-              is GenericAuthViewModel.Event.SubmitEmailSuccess -> startOtpInputActivity(event)
-            }
-          }
-        }
+      viewState.otpId?.let {
+        model.onStartOtpInput()
+        startOtpInputActivity(otpId = it, email = viewState.input)
       }
```

Comment:
A slight concern about doing a start activity side effect in composition.
I think we need to replace this with something like:
```suggestion
      LaunchedEffect(viewState.otpId) {
          val otpId = viewState.otpId ?: return@LaunchedEffect
          model.onStartOtpInput()
          startOtpInputActivity(otpId = otpId, email = viewState.input)
      }
```
That way we're not running a startActivity in composition which sounds a bit scary, I am not sure if there will ever be a case where two of them will run before model.onStartOtpInput is run and finished.

Also a good section to read https://developer.android.com/jetpack/compose/performance#avoid-backwards
Here we're reading viewState.otpId in composition, and then we're calling onStartOtpInput which changes the otpId since it changes the state in the VM. And we're not doing this inside any sort of effect handler like LaunchedEffect.